sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/DataFeedLiveTests.cs

```diff
@@ -2435,6 +2041,151 @@ private void ValidateGenericDataSource(DataFeedSource dataSource, bool isAdmin)
+            {
+                var concreteDataSource = dataSource as InfluxDbDataFeedSource;
+
+                Assert.That(concreteDataSource, Is.Not.Null);
+                Assert.That(concreteDataSource.DataSourceKind, Is.EqualTo(DataFeedSourceKind.InfluxDb));
+                Assert.That(concreteDataSource.Database, Is.EqualTo(DataSourceDatabase));
+                Assert.That(concreteDataSource.Username, Is.EqualTo(DataSourceUsername));
+                Assert.That(concreteDataSource.Query, Is.EqualTo(DataSourceQuery));
+            }
+            else if (expectedKind == nameof(DataFeedSourceKind.LogAnalytics))
+            {
+                var concreteDataSource = dataSource as LogAnalyticsDataFeedSource;
+
+                Assert.That(concreteDataSource, Is.Not.Null);
+                Assert.That(concreteDataSource.DataSourceKind, Is.EqualTo(DataFeedSourceKind.LogAnalytics));
+                Assert.That(concreteDataSource.WorkspaceId, Is.EqualTo(DataSourceWorkspaceId));
+                Assert.That(concreteDataSource.Query, Is.EqualTo(DataSourceQuery));
+                Assert.That(concreteDataSource.ClientId, Is.EqualTo(DataSourceClientId));
+                Assert.That(concreteDataSource.TenantId, Is.EqualTo(DataSourceTenantId));
+            }
+            else if (expectedKind == nameof(DataFeedSourceKind.MongoDb))
+            {
+                var concreteDataSource = dataSource as MongoDbDataFeedSource;
+
+                Assert.That(concreteDataSource, Is.Not.Null);
+                Assert.That(concreteDataSource.DataSourceKind, Is.EqualTo(DataFeedSourceKind.MongoDb));
+                Assert.That(concreteDataSource.Database, Is.EqualTo(DataSourceDatabase));
+                Assert.That(concreteDataSource.Command, Is.EqualTo(DataSourceCommand));
+            }
+            else if (expectedKind == nameof(DataFeedSourceKind.MySql))
+            {
+                var concreteDataSource = dataSource as MySqlDataFeedSource;
+
+                Assert.That(concreteDataSource, Is.Not.Null);
+                Assert.That(concreteDataSource.DataSourceKind, Is.EqualTo(DataFeedSourceKind.MySql));
+                Assert.That(concreteDataSource.Query, Is.EqualTo(DataSourceQuery));
+            }
+            else if (expectedKind == nameof(DataFeedSourceKind.PostgreSql))
+            {
+                var concreteDataSource = dataSource as PostgreSqlDataFeedSource;
+
+                Assert.That(concreteDataSource, Is.Not.Null);
+                Assert.That(concreteDataSource.DataSourceKind, Is.EqualTo(DataFeedSourceKind.PostgreSql));
+                Assert.That(concreteDataSource.Query, Is.EqualTo(DataSourceQuery));
+            }
+            else if (expectedKind == nameof(DataFeedSourceKind.SqlServer))
+            {
+                var concreteDataSource = dataSource as SqlServerDataFeedSource;
+
+                Assert.That(concreteDataSource, Is.Not.Null);
```

Comment:
could we call the specific `ValidateSqlServerDataSource` method here instead of adding extra validation? or is there a reason why we can't?

Reply:
I'm just keeping the `Validate<data-source-name>DataSource` methods in this PR because they are still being used by the Update methods. I'll remove all of the specific methods in my next PR since will do the same we did here for Update methods.